cli/setup.go

```diff
@@ -154,12 +154,22 @@ const (
 // ---------------------------- END Setup constants ----------------------------
 
 func getDefaultDeviceType() string {
-	hostName, err := ioutil.ReadFile("/etc/hostname")
+	var devType string
+	deviceType, err := ioutil.ReadFile(conf.DefaultDeviceTypeFile)
 	if err != nil {
-		return "unknown"
+		hostName, err := ioutil.ReadFile("/etc/hostname")
+		if err != nil {
+			return "unknown"
+		}
+		devType = string(hostName)
+		devType = strings.Trim(devType, "\n")
+	} else {
+		devType = string(deviceType)
+		idx := strings.Index(devType, "=")
+		if idx >= 0 && len(deviceType) > idx {
+			devType = devType[idx+1:]
+		}
```

Comment:
I think you can reuse [GetDeviceType](https://github.com/mendersoftware/mender/blob/master/device/device.go#L143) from `device.go`.